### PR TITLE
CLDC-2494 Add duplicate logs page

### PR DESCRIPTION
**Pull request by Airk0n**
Needs https://github.com/communitiesuk/submit-social-housing-lettings-and-sales-data/pull/1762 to correctly determine duplicates.

This PR adds a duplicate logs controller with the page that displays all the duplicates for a specific log.
image
image

### CODE REVIEW

File: app/controllers/duplicate_logs_controller.rb [hunk 1]
```diff
@@ -0,0 +1,27 @@
+class DuplicateLogsController < ApplicationController
+  before_action :find_resource_by_named_id
+
+  def show
+    @duplicate_logs = @log.class.duplicate_logs_for_organisation(current_user.organisation_id, @log)
+    @all_duplicates = [@log, *@duplicate_logs]
+    @duplicate_check_questions = duplicate_check_question_ids.map { |question_id| @log.form.get_question(question_id, @log) }.compact
```
Comment (ghost) [inline app/controllers/duplicate_logs_controller.rb]: I feel all this + `duplicate_check_question_ids` could be in an helper or presenter model/module where we could unit test it

File: app/controllers/form_controller.rb [hunk 2]
```diff
@@ -155,6 +155,10 @@ def previous_interruption_screen_referrer
   end
 
   def successful_redirect_path
+    if @log.class.duplicate_logs_for_organisation(current_user.organisation, @log).count.positive?
```
Comment (ghost) [inline app/controllers/form_controller.rb]: this change is not unit tested

Comment (kosiakkatrina) [reply app/controllers/form_controller.rb]: Hmm, we're testing that the redirect happens when there are duplicate logs in form_controller_spec, and the opposite condition would get tested by other successful redirect cases that expect some of the further code to get executed. I feel like that covers it (?)

File: app/controllers/duplicate_logs_controller.rb (hunk 1 above)
Comment (kosiakkatrina) [reply app/controllers/duplicate_logs_controller.rb]: Hm, so the scopes are tested here, the one in this PR is just a placeholder https://github.com/communitiesuk/submit-social-housing-lettings-and-sales-data/pull/1762
And the rest is tested in request specs - we're checking that correct logs and correct questions get displayed

**Review by ghost — APPROVED** [review]
@kosiakkatrina thanks for replying to my comments, all good!

**Review by ghost — APPROVED** [review]

**Pull request merged**